jwst/combine_1d/combine1d.py

```diff
@@ -199,7 +200,12 @@ def accumulate_sums(self, input_spectra):
                 if in_spec.dq[i] & datamodels.dqflags.pixel['DO_NOT_USE'] > 0:
                     continue
                 # Round to the nearest pixel.
+                if np.isnan(out_pixel[i]):
```

Comment:
Is there a way to do the search for NaN on the entire out_pixel array at once (i.e. outside of this loop over i), rather than having to do the test on every loop iteration? Or would waiting to do the check after the loop is done create problems for the calculations being done in the remainder of the loop? If it's not a big performance hit, just leave it as is.

Reply:
Yes, `np.isnan` can operate on an array and return a boolean array, but I would still need to check an element of that boolean array inside the loop.  I thought the check on NaN would be quick, but checking a boolean value has to be quicker.  I'll take the call to `np.isnan` out of the loop.